e2e/auth/login.testSuite.ts

```diff
@@ -54,6 +65,59 @@ export const authLoginTestSuite = () => {
         .expect(200);
     });
 
+    it('should not authorize user if request origin does not matches origin of token', async () => {
+      const identityToken = await getIdentityToken(provider, wallet);
+
+      const loginResponse = await request(app.getHttpServer())
+        .post('/v1/login')
+        .set('Origin', allowedOrigins[0])
+        .send({
+          identityToken,
+        })
+        .expect(201);
+
+      return request(app.getHttpServer())
+        .get('/v1/search/test')
+        .set('Origin', allowedOrigins[1])
+        .set('Cookie', [
+          loginResponse.headers['set-cookie'][0].split(';')[0] + ';',
+          loginResponse.headers['set-cookie'][1].split(';')[0] + ';',
+        ])
+        .expect(401)
+        .expect(function (res) {
+          expect(res.text).toContain(
+            `Token origin ${allowedOrigins[0]} does not matches request origin ${allowedOrigins[1]}`
+          );
+        });
+    });
+
+    it('should not login from not allowed origin', async () => {
+      const identityToken = await getIdentityToken(provider, wallet);
+      await request(app.getHttpServer())
+        .post('/v1/login')
+        .set('origin', notAllowedOrigin)
+        .send({
+          identityToken,
+        })
+        .expect(401)
+        .expect(function (res) {
+          expect(res.text).toContain(
+            `Origin ${notAllowedOrigin} is not allowed`
+          );
+        });
+    });
+
+    it('should login from allowed origin', async () => {
+      const identityToken = await getIdentityToken(provider, wallet);
+      await request(app.getHttpServer())
+        .post('/v1/login')
+        .set('origin', allowedOrigins[0])
+        .send({
+          identityToken,
+        })
+        .expect(201);
+    });
+
```

Comment:
We have one test when origin header isn't set https://github.com/energywebfoundation/ssi-hub/blob/44b16b3694f27773b0ea9ba02b26e6fe295ba657/e2e/auth/login.testSuite.ts#L35. But I agree that some more can be added. Grouping tests when origin is set or not will also help.
By the way from my opinion request to `v1/search/test` in your test should fail, because `Origin` header doesn't match origin of access token, which undefined. Why we should allow this request?

Reply:
Ah, thanks, I didn't see the test that you linked to. I think that test is probably enough. In fact, the test you linked to is better because it is retrieving the token from the cookie header in the response and not from the response body (see https://energyweb.atlassian.net/browse/SWTCH-2020)